packages/data-stores/src/products-list/actions.ts

```diff
@@ -0,0 +1,53 @@
+import { ensureNumericCost } from './assembler';
+import {
+	PRODUCTS_LIST_RECEIVE,
+	PRODUCTS_LIST_REQUEST,
+	PRODUCTS_LIST_REQUEST_FAILURE,
+} from './constants';
+import type { ProductsList, ProductsListFailure, ProductsListResponse } from './types';
+
+export function createActions() {
+	const receiveProductsList = (
+		productsList: ProductsList,
+		type: string | null = null
+	): ProductsListResponse => {
+		// Since the request succeeded, productsList should be guaranteed non-null;
+		// thus, we don't have any safety checks before this line.
+
+		// Create a completely new products list to avoid mutating the original
+		const sanitizedProductsList = Object.fromEntries(
+			Object.entries( productsList ).map( ( [ slug, product ] ) => [
+				slug,
+				ensureNumericCost( product ),
+			] )
+		);
+
+		return {
+			type: PRODUCTS_LIST_RECEIVE,
```

Comment:
This can be written as such:
```suggestion
			type: 'PRODUCTS_LIST_RECEIVE' as const // the "as const" tells TS this isn't a variable, to it can be treated as a type, thus preventing mistakes
```